src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,529 @@
+import { ContentTypeElements } from "@kontent-ai/management-sdk";
+import chalk from "chalk";
+import { readFileSync, writeFileSync } from "fs";
+import open from "open";
+import { resolve } from "path";
+
+import { SyncParams } from "../../commands/diff/diff.js";
+import { logError, logInfo } from "../../log.js";
+import {
+  isContentTypeData,
+  isContentTypeSnippetData,
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isTaxonomyRequestModel,
+  isValidationRegex,
+} from "../../utils/typeguards.js";
+import { RequiredCodename } from "../../utils/types.js";
+import { DiffModel, DiffObject } from "./types/diffModel.js";
+import { PatchOperation, ReplacePatchOperationValue } from "./types/patchOperation.js";
+
+type DiffData = DiffModel & SyncParams;
+type EntityPathHandler = {
+  regex: RegExp;
+  entity: (() => string) | ((match: string[]) => string);
+};
+
+export const writeDiffToFile = (diffData: DiffData) => {
+  const template = readFileSync(resolve("./", "diffTemplate.html"), "utf-8");
+  const resolvedTemplate = template.replace(
+    /{{.*?}}/g,
+    (match) => templateHandlerMap.get(match)?.(diffData) ?? match,
+  );
+  const outputPath = resolve(
+    diffData.outPath!,
+    `diff_${new Date().toISOString().replace(/[:.-]/g, "_")}.html`,
+  );
+
+  try {
+    logInfo(
+      diffData,
+      "standard",
+      chalk.yellow(`Generating a diff file at ${outputPath}`),
+    );
+    writeFileSync(outputPath, resolvedTemplate);
```

Comment:
agreed. adjusted logic to create missing path segments recursively and take into account if the path includes `.html` suffix to use that for the name instead of the generic one.